traitsui/extras/tests/test_demo.py

```diff
@@ -0,0 +1,144 @@
+#  Copyright (c) 2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+
+import os
+import tempfile
+import textwrap
+import unittest
+from xml.etree import ElementTree as ET
+
+from traitsui.extras import demo
+
+HTML_NS_PREFIX = "{http://www.w3.org/1999/xhtml}"
+
+
+def get_html_tag(tag):
+    return HTML_NS_PREFIX + tag
+
+
+class TestDemoPathDescription(unittest.TestCase):
+    """ Test ``DemoPath.description`` """
+
+    def test_description_with_empty_directory(self):
+        # If the directory is empty, the content of the description should
+        # be empty.
+        with tempfile.TemporaryDirectory() as directory:
+            model = demo.DemoPath(
+                name=directory,
+            )
+
+            tree = ET.fromstring(model.description)
+            body_node = next(tree.iter(get_html_tag("body")))
+            div_node, = list(body_node)
+            self.assertEqual(list(div_node), [])
+
+    def test_use_index_rst(self):
+        with tempfile.TemporaryDirectory() as directory:
+            index_rst = os.path.join(directory, "index.rst")
+            with open(index_rst, "w", encoding="utf-8") as f:
+                f.write(".. image:: any_image.jpg\n")
+
+            model = demo.DemoPath(
+                name=directory,
+            )
+
+            tree = ET.fromstring(model.description)
+            img_node = next(tree.iter(get_html_tag("img")))
+            self.assertEqual(img_node.attrib["src"], "any_image.jpg")
+
+    def test_description_use_css(self):
+        with tempfile.TemporaryDirectory() as directory:
+            model = demo.DemoPath(
+                name=directory,
+                css_filename="default.css",
+            )
+
+            tree = ET.fromstring(model.description)
+        link_node = next(tree.iter(get_html_tag("link")))
+        self.assertEqual(link_node.attrib["href"], "default.css")
```

Comment:
Thank you. I moved the assertions in the other tests out of the context: By the time the XML tree is parsed, we don't need the temporary directory any more. The tests should be more consistent now.

Reply:
Actually, no, there are still more tests in `TestDemoPathInitLocals` that have the assertions in the context. 
I wonder how the application behaves if the directory is deleted half-way through running it... it probably won't deal with this very well. I think I will move the assertions back in actually, because it is closer to what the application assumes.

Reply:
>  I think I will move the assertions back in actually, because it is closer to what the application assumes.

Going back on that again 😂 . The XML tree is not part of the application data any more so that one is reasonable to stay out of the context. Sorry.